### DISCUSSION TITLE
Custom diff

### DISCUSSION THREAD

**Discussion by githubkusi** [gitextensions/gitextensions#8209] [General]
I've created a custom script "mydiff" and made sure its on the path environment variable

    #!/bin/bash
    echo "my diff"

My .gitattributes

    *.mat diff=mydiff

Adding the custom diff to the repo

    $git config diff.mydiff.command mydiff

In my git repo, I have a changed file "myfile.mat". On the command line (Console Tab in GE), I get the following diff

    $ git diff
    my diff

The output is as expected from the script *mydiff* above. However, in the diff pane of the commit window in GE, I get nothing, just a blank pane. Is this a bug of GE or intended or am I doing something wrong?

**Comment by gerhardol** [comment]
GE expects the output in diff format, to be able to parse the output for patches etc
So this script fail

(Overriding the general diff command will affect listing of files etc too)

**Reply by githubkusi** [reply]
Could I fake the output of my diff script such that it resembles the diff format?

To give a bit more context for my request: I want to visualize the difference between two hdf5 files
[h5diff manpage](https://support.hdfgroup.org/HDF5/doc/RM/Tools.html#Tools-Diff)

**Reply by gerhardol** [reply]
To see the Git command GE creates, see the command log (F12)
See ```CreatePatchesFromString``` for the parsing, if you are in doubt. There are likely tests as an example too

Line patching etc will fail of course
